packages/eas-cli/src/project/ios/exemptEncryption.ts

```diff
@@ -0,0 +1,196 @@
+import chalk from 'chalk';
+import fs from 'fs-extra';
+import path from 'node:path';
+
+import { readAppJson } from '../../build/utils/appJson';
+import Log, { learnMore } from '../../log';
+import { promptAsync } from '../../prompts';
+import { Client } from '../../vcs/vcs';
+import { getProjectConfigDescription } from '../projectUtils';
+import { resolveWorkflowAsync } from '../workflow';
+
+/** Non-exempt encryption must be set on every build in App Store Connect, we move it to before the build process to attempt only setting it once for the entire life-cycle of the project. */
+export async function ensureNonExemptEncryptionIsDefinedForManagedProjectAsync({
+  projectDir,
+  exp,
+  vcsClient,
+  nonInteractive,
+}: {
+  projectDir: string;
+  exp: ExpoConfig;
+  vcsClient: Client;
+  nonInteractive: boolean;
+}): Promise<void> {
+  const workflow = await resolveWorkflowAsync(projectDir, Platform.IOS, vcsClient);
+  assert(workflow === Workflow.MANAGED, 'This function should be called only for managed projects');
+
+  try {
+    await getNonExemptEncryptionAsync(projectDir, exp, vcsClient);
+  } catch {
+    await configureNonExemptEncryptionAsync({
+      projectDir,
+      exp,
+      nonInteractive,
+    });
+  }
+}
+
+export class AmbiguousNonExemptEncryptionError extends Error {
+  constructor(message?: string) {
+    super(message ?? 'Could not resolve non-exempt encryption setting.');
+  }
+}
+
+export async function getNonExemptEncryptionAsync(
+  projectDir: string,
+  exp: ExpoConfig,
+  vcsClient: Client,
+  xcodeContext?: { targetName?: string; buildConfiguration?: string }
+): Promise<void> {
+  const workflow = await resolveWorkflowAsync(projectDir, Platform.IOS, vcsClient);
+
+  if (workflow === Workflow.GENERIC) {
+    const xcodeProject = IOSConfig.XcodeUtils.getPbxproj(projectDir);
+
+    const infoPlistBuildProperty = getInfoPlistPathFromPbxproj(xcodeProject);
+    if (!infoPlistBuildProperty) {
+      throw new Error('Info.plist file linked to Xcode project does not exist');
+    }
+
+    //: [root]/myapp/ios/MyApp/Info.plist
+    const infoPlistPath = path.join(
+      //: myapp/ios
+      projectDir,
+      'ios',
+      //: MyApp/Info.plist
+      infoPlistBuildProperty
+    );
+    if (!(await fs.promises.stat(infoPlistPath))) {
+      throw new Error(`Info.plist file linked to Xcode project does not exist: ${infoPlistPath}`);
+    }
+    const infoPlist = plist.parse(await fs.promises.readFile(infoPlistPath, 'utf8'));
+
+    const nonExemptEncryption = infoPlist['ITSAppUsesNonExemptEncryption'];
+    const buildConfigurationDesc =
+      xcodeContext?.targetName && xcodeContext?.buildConfiguration
+        ? ` (target = ${xcodeContext.targetName}, build configuration = ${xcodeContext.buildConfiguration})`
+        : '';
+    assert(
+      nonExemptEncryption !== undefined,
+      `Could not read non-exempt encryption setting from Xcode project${buildConfigurationDesc}.`
+    );
+    return nonExemptEncryption;
```

Comment:
if we run it only for managed does it mean we should delete it? or maybe we should do it for generic as well?

I guess that we should do it for generic as well, right?